mir-ci/mir_ci/test_tests.py

```diff
@@ -50,10 +55,103 @@ async def test_program_can_be_terminated(self) -> None:
 
     async def test_program_is_killed_when_terminate_fails(self) -> None:
         start = time.time()
-        p = Program(['sh', '-c', 'trap "" TERM; echo abc; sleep 1; echo ijk; sleep 5; echo xyz'])
+        p = Program(App(['sh', '-c', 'trap "" TERM; echo abc; sleep 1; echo ijk; sleep 5; echo xyz']))
         async with p:
             await asyncio.sleep(0.5)
             await p.kill(2)
         elapsed = time.time() - start
         assert p.output.strip() == 'abc\nijk'
         assert abs(elapsed - 2.5) < 0.1
+
+    @patch("uuid.uuid4")
+    async def test_program_runs_with_systemd_when_flag_is_set(self, mock_uuid) -> None:
+        mock_uuid.return_value = "12345"
+        start = time.time()
+        p = Program(App(['sh', '-c', 'sleep 1'], "deb"))
+        async with p:
+            await asyncio.sleep(0.5)
+            await p.kill(2)
+        mock_uuid.assert_called_once()
+
+
+@pytest.mark.self
+class TestBenchmarker:
+    @staticmethod
+    def create_program_mock():
+        def async_return():
+            f = asyncio.Future()
+            f.set_result(MagicMock())
+            return f
+        p = MagicMock()
+        p.get_cgroup = Mock(return_value=async_return())
+        return p
+    
+    async def test_benchmarker_with_program(self) -> None:
+        p = TestBenchmarker.create_program_mock()
+        benchmarker = Benchmarker({"program": p}, poll_time_seconds=0.1)
+        async with benchmarker:
+            await asyncio.sleep(1)
+
+        p.get_cgroup.assert_called()
+        p.__aenter__.assert_called_once()
+        p.__aexit__.assert_called_once()
+
+    async def test_benchmarker_can_generate_report(self) -> None:
+        p = TestBenchmarker.create_program_mock()
+        benchmarker = Benchmarker({"program": p}, poll_time_seconds=0.1)
+        async with benchmarker:
+            await asyncio.sleep(1)
+
+        callback = Mock()
+        benchmarker.generate_report(callback)
+        callback.assert_called()
+
+    async def test_benchmarker_cant_enter_twice(self) -> None:
+        p = TestBenchmarker.create_program_mock()
+        benchmarker = Benchmarker({"program": p}, poll_time_seconds=0.1)
+        async with benchmarker:
+            async with benchmarker:
+                await asyncio.sleep(1)
+        
+        p.__aenter__.assert_called_once()
+
+@pytest.mark.self
+class TestCgroup:
+    @patch("builtins.open", new_callable=mock_open, read_data="usage_usec 100")
+    def test_cgroup_can_get_cpu_time_microseconds(self, mock_open):
+        cgroup = Cgroup(12345, "/fake/path")
+        assert(cgroup.get_cpu_time_microseconds() == 100)
```

Comment:
```suggestion
        self.assertEqual(cgroup.get_cpu_time_microseconds(), 100)
```

Maybe add a couple more lines to the `read_data` so we're sure it iterates over more lines?

Reply:
What is this `self.assertEqual`? I don't see it used anywhere else and python seems to not like it.